src/zope/app/container/browser/tests/test_contents_functional.py

```diff
@@ -305,40 +288,46 @@ def test_readonly_display(self):
         root['foo'] = ReadOnlyContainer()
         transaction.commit()
         response = self.publish('/foo/@@contents.html', basic='mgr:mgrpw')
-        self.assertEqual(response.getStatus(), 200)
+        self.assertEqual(response.status_int, 200)
 
     def test_uncopyable_object(self):
-        ztapi.provideAdapter(IUncopyable,
-                             copypastemove.interfaces.IObjectCopier,
-                             ObjectNonCopier)
+        provideAdapter(IUncopyable,
+                       copypastemove.interfaces.IObjectCopier,
+                       ObjectNonCopier)
         root = self.getRootFolder()
         root['uncopyable'] = UncopyableFile()
         transaction.commit()
         response = self.publish('/@@contents.html',
                                 basic='mgr:mgrpw',
-                                form={'ids': [u'uncopyable'],
+                                form={'ids:list': [u'uncopyable'],
                                       'container_copy_button': u'Copy'})
-        self.assertEqual(response.getStatus(), 200)
-        body = response.getBody()
-        self.assert_("cannot be copied" in body)
+        self.assertEqual(response.status_int, 200)
+        body = response.text
+        self.assertIn("cannot be copied", body)
 
     def test_unmoveable_object(self):
-        ztapi.provideAdapter(IImmovable,
-                             copypastemove.interfaces.IObjectMover,
-                             ObjectNonMover)
+        provideAdapter(IImmovable,
+                       copypastemove.interfaces.IObjectMover,
+                       ObjectNonMover)
         root = self.getRootFolder()
         root['immovable'] = ImmovableFile()
         transaction.commit()
         response = self.publish('/@@contents.html',
                                 basic='mgr:mgrpw',
-                                form={'ids': [u'immovable'],
+                                form={'ids:list': [u'immovable'],
                                       'container_cut_button': u'Cut'})
-        self.assertEqual(response.getStatus(), 200)
-        body = response.getBody()
-        self.assert_("cannot be moved" in body)
+        self.assertEqual(response.status_int, 200)
+        body = response.text
+        self.assertIn("cannot be moved", body)
 
+    @unittest.skipIf(str is not bytes, #Py2 only
+                     "Only Python 2 can do str(b'encoded') and get the right thing")
     def test_copy_then_delete_with_unicode_name(self):
-        """Tests unicode on object copied then deleted (#238579)."""
+        # Tests unicode on object copied then deleted (#238579)
+        # The zope.publisher.browser conversion methods for text/str
+        # rely on the default encoding, which breaks on this unicode name in Python 3.
+        # We either wind up with "b'voil\\xe0'" or a list of the ints that make
+        # up the bytes, depending on which order we try for the type names.
```

Comment:
This is not great :/

I'm fine with filing a separate bug, "Unicode object names are broken on Python 3" instead of postponing this PR.  I would, however, prefer for that bug to be fixed before making a zope.app.container release that claims Python 3 support.